semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/ClasspathLookup.scala

```diff
@@ -0,0 +1,134 @@
+package scala.meta.internal.metacp
+
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.jar.JarFile
+import scala.collection.mutable
+import scala.meta.io.AbsolutePath
+import scala.meta.io.Classpath
+import scala.meta.internal.io.PathIO
+
+final class ClasspathLookup private (val dirs: collection.Map[String, ClasspathEntry.Package]) {
+
+  override def toString: String = {
+    s"ClasspathLookup(${dirs.size} entries)"
+  }
+
+  def getEntry(path: String): Option[ReadableClasspathEntry] = {
```

Comment:
If you accept the naming suggestions from above, it would be consistent to rename this method to `getClassfile`.